floodlight/io/datasets.py

```diff
@@ -799,3 +801,278 @@ def _download_matches_info(self) -> None:
+                    self._IDSSE_FILE_NAME_INFO, self._IDSSE_HOST_URL_INFO
+                )
+            if not os.path.isfile(f"{self._data_dir}/{self._IDSSE_FILE_NAME_EVENT}"):
+                self._download_and_write(
+                    self._IDSSE_FILE_NAME_EVENT, self._IDSSE_HOST_URL_EVENT
+                )
+            if not os.path.isfile(f"{self._data_dir}/{self._IDSSE_FILE_NAME_POSITION}"):
+                self._download_and_write(
+                    self._IDSSE_FILE_NAME_POSITION, self._IDSSE_HOST_URL_POSITION
+                )
+        elif match_id == "all":
+            for file_id in self._IDSSE_FILE_IDS_INFO:
+                if file_id in ["J03WMX", "J03WN1"]:
+                    competition = "DFL-COM-000001"
+                else:
+                    competition = "DFL-COM-000002"
+                self._IDSSE_HOST_URL_INFO = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_INFO[file_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
+                self._IDSSE_HOST_URL_EVENT = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_EVENT[file_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
+                self._IDSSE_HOST_URL_POSITION = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_POSITION[file_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
+
+                self._IDSSE_FILE_NAME_INFO = f"DFL_02_01_matchinformation_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+                self._IDSSE_FILE_NAME_EVENT = f"DFL_03_02_events_raw_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+                self._IDSSE_FILE_NAME_POSITION = f"DFL_04_03_positions_raw_observed_{competition}_DFL-MAT-{file_id}.{self._IDSSE_FILE_EXT}"
+
+                if not os.path.isfile(f"{self._data_dir}/{self._IDSSE_FILE_NAME_INFO}"):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_INFO, self._IDSSE_HOST_URL_INFO
+                    )
+                if not os.path.isfile(
+                    f"{self._data_dir}/{self._IDSSE_FILE_NAME_EVENT}"
+                ):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_EVENT, self._IDSSE_HOST_URL_EVENT
+                    )
+                if not os.path.isfile(
+                    f"{self._data_dir}/{self._IDSSE_FILE_NAME_POSITION}"
+                ):
+                    self._download_and_write(
+                        self._IDSSE_FILE_NAME_POSITION, self._IDSSE_HOST_URL_POSITION
+                    )
+
+    def get(
+        self,
+        match_id: str = "J03WMX",
+        teamsheet_home: Teamsheet = None,
+        teamsheet_away: Teamsheet = None,
+        events=True,
+        positions=True,
+    ) -> Tuple[
+        tuple[dict[str, dict[str, Events]], dict[str, Teamsheet], Pitch],
+        tuple[
+            dict[str, dict[str, XY]],
+            dict[str, Code],
+            dict[str, Code],
+            dict[str, Teamsheet],
```

Comment:
is that a new convention I missed that the second nested "tuple" and "dict" are lowercase?

Reply:
It's probably not... Fixed